src/TLuaInterpreter.cpp

```diff
@@ -192,6 +192,77 @@ TLuaInterpreter::~TLuaInterpreter()
     lua_close(pGlobalLua);
 }
 
+// No documentation available in wiki - internal function
+// Replaces a check like this:
+//    if (!lua_isboolean(L, 9)) {
+//        lua_pushfstring(L,
+//            "createMapLabel: bad argument #9 type (showOnTop as number expected, got %s!)",
+//            luaL_typename(L, 1));
+//        return lua_error(L);
+//    }
+//    bool showOnTop = lua_toboolean(L, 9);
+//
+// With reduced repetition like that:
+//    bool showOnTop = verifyBoolean(L, 9, "createMapLabel", "showOnTop");
+//
+// The "isOptional" parameter is optional, and will default to not-optional parameters! :)
+//
+// See also: verifyString, verifyInt, verifyFloat, announceWrongArgumentType
+bool TLuaInterpreter::verifyBoolean(lua_State* L, const int pos, const QString& functionName, const QString& publicName, const bool isOptional)
```

Comment:
As the strings are squirted straight into a `lua_pushfstring(...)` you do not need to dress them up as `QString`s - `const char*` is perfectly fine in this *particular* circumstances...

Reply:
That might help with all those errors further down! :stuck_out_tongue_winking_eye:

Reply:
Which errors are you referring? Is this wrong or merely inconvenient? 
I for one know QString better than char* so I used what I knew already.

Reply:
You cannot use a `QString` as a parameter to go into a `%s` in a `lua_pushfstring(...)` call, you need to convert it back to a `const char*` with a `QString::toUft()` followed by a `QByteArray::constData()` call - so you might just as well avoid a round trip from `const char*` ==> `QString` ==> `const char*` ...